src/Generator/PdfOptionsGenerator.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\InvoicingPlugin\Generator;
+
+use Symfony\Component\Config\FileLocatorInterface;
+
+final class PdfOptionsGenerator implements PdfOptionsGeneratorInterface
+{
+    public function __construct(
+        private FileLocatorInterface $fileLocator,
+        private array $knpSnappyOptions,
+        private array $allowedFiles
+    ) {
+    }
+
+    public function generate(): array
+    {
+        $options = $this->knpSnappyOptions;
+
+        if (empty($this->allowedFiles)) {
+            return $options;
+        }
+
+        if (!isset($options['allow'])) {
+            $options['allow'] = [];
+        } elseif (!is_array($options['allow'])) {
+            $options['allow'] = [$options['allow']];
+        }
+
+        $options['allow'] = array_merge(
+            $options['allow'],
+            array_map(fn ($file) => $this->fileLocator->locate($file), $this->allowedFiles)
+        );
+
+        return $options;
```

Comment:
Just a style suggestion:

```suggestion
        $locatedFiles = array_map(fn ($file) => $this->fileLocator->locate($file), $this->allowedFiles);
        
        if (!isset($options['allow'])) {
            return $locatedFiles;
        } 
        
        if (!is_array($options['allow'])) {
            $options['allow'] = [$options['allow']];
        }

        $options['allow'] = array_merge(
            $options['allow'],
            $locatedFiles
        );

        return $options;
```

However, some cases are not covered by this class spec. For sure we do not cover the absence of `$options['allow']`. The case with `$options['allow']` as the array is not covered as well. When can it happen? 

It is not the case for spec probably, but what will happen if the file does not exist? What is returned by the fileLocator in such a case?

Reply:
Now `$options['allow']` cannot be an array but it could be possible in the future because it makes sense so I have put this if statement to cover that case anyway :thinking: 

File locator throws an error for unexisting files.